codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/transformers/ShapesReachableFromOperationInputTagger.kt

```diff
@@ -52,7 +54,7 @@ object ShapesReachableFromOperationInputTagger {
 
         return ModelTransformer.create().mapShapes(model) { shape ->
             when (shape) {
-                is StructureShape, is UnionShape, is ListShape, is MapShape, is StringShape, is IntegerShape, is ShortShape -> {
+                is StructureShape, is UnionShape, is ListShape, is MapShape, is StringShape, is IntegerShape, is ShortShape, is LongShape, is ByteShape -> {
                     if (shapesReachableFromOperationInputs.contains(shape)) {
                         val builder = when (shape) {
```

Comment:
Do we need this `when` at all? It seems that it covers the same cases as the encompassing `is` branch.

Reply:
I think it's necessary for type refinement, otherwise you have no way to call `toBuilder`.

Reply:
You're right, I thought `toBuilder()` was defined on `Shape`. My bad!